src/filters/qualitative-filter.js

```diff
@@ -0,0 +1,221 @@
+import React, { useMemo, useState, useRef, useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+import List from '@material-ui/core/List'
+import ListItem from '@material-ui/core/ListItem'
+import ListItemIcon from '@material-ui/core/ListItemIcon'
+import ListItemText from '@material-ui/core/ListItemText'
+import { makeStyles } from '@material-ui/core/styles'
+import TextField from '@material-ui/core/TextField'
+import InputAdornment from '@material-ui/core/InputAdornment'
+import SearchIcon from '@material-ui/icons/Search'
+import { Button, Checkbox } from '@material-ui/core'
+
+
+const useStyles = makeStyles(() => ({
+  list: {
+    width: '20em',
+    overflow: 'auto',
+    maxHeight: '60vh',
+    padding: '0 1.5em 0 1em',
+  },
+
+  listItemContainer: {
+    display: 'flex',
+  },
+
+  MuiListItem: {
+    paddingLeft: 10,
+  },
+
+  MuiListItemIcon: {
+    minWidth: 0,
+  },
+
+  onlyButton: {
+    textTransform: 'capitalize',
+    display: 'none',
+  },
+
+  buttonContainer: {
+    marginTop: 20,
+    display: 'grid',
+    'grid-template-columns': '1fr 1fr',
+    columnGap: 10,
+  },
+}))
+
+const QualitativeFilter = ({ column: { filterValue, preFilteredRows, setFilter, id }, closePopper }) => {
+  const [value, setValue] = useState('')
+  const [optionsValue, setOptionsValue] = useState(filterValue || '')
+  const listItemRef = useRef(null)
+
+  let listContainer = ''
+  let li = ''
+  let onlyButtonContainer = ''
+
+  const classes = useStyles()
+
+  const options = useMemo(() => {
+    const opts = new Set()
+    preFilteredRows.forEach((row) => {
+      opts.add(row.values[id])
+    })
+    return [...opts.values()]
+  }, [id, preFilteredRows])
+
+  useEffect(() => {
+    if (listItemRef.current !== null) {
+      listContainer = listItemRef.current.getElementsByClassName(classes.listItemContainer)
+      li = listItemRef.current.getElementsByClassName('listItemText')
+      onlyButtonContainer = listItemRef.current.getElementsByClassName(classes.onlyButton)
```

Comment:
I think it'd be great if we can incorporate react way of dom manipulation as using a vanila DOM API could cause unexpected problems later on.